### PR TITLE
V22F-346 remove excess menus

### PR DESCRIPTION
**Pull request by sjoeman1**
Removed every menu that we will no longer use. For now the race menus are mostly untouched, except for networking menus, since we still need to use those, the client probably not though.

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/confscreens/controlconfig.cpp [hunk 1]
```diff
@@ -3,30 +3,31 @@
...
+  *   the Free Software Foundation; either version 2 of the License, or     *
+  *   (at your option) any later version.                                   *
+  *                                                                         *
+  ***************************************************************************/
+
+  /** @file
+              Human player control configuration menu
+      @author	<a href=mailto:eric.espie@torcs.org>Eric Espie</a>
+      @version	$Id: controlconfig.cpp 6966 2020-04-25 17:33:03Z scttgs0 $
+  */
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/confscreens/controlconfig.cpp:24]: Please do not run the automatic linter on a speed-dreams file.

Because of this I cannot really find which changes were *actually* made, because almost every line his changed due to removal of tabs, moving pointer from variable to type, etc, etc.

Is it possible to revert the linted file and then add your changes to that?

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 2]
```diff
@@ -8,19 +8,20 @@
...
- *                                                                         *
- ***************************************************************************/
+ /***************************************************************************
+  *                                                                         *
+  *   This program is free software; you can redistribute it and/or modify  *
+  *   it under the terms of the GNU General Public License as published by  *
+  *   the Free Software Foundation; either version 2 of the License, or     *
+  *   (at your option) any later version.                                   *
+  *                                                                         *
+  ***************************************************************************/
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:18]: Don't run the linter on a speed-dreams file, please revert this and then add your changes normally.

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 3]
```diff
@@ -29,58 +30,59 @@
 #include "DataSelectionMenu.h"
 
 
-static void *MenuHandle = 0;
+static void* MainMenuHandle = 0;
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:33]: You need to state that this variable name was updated, here and in CHANGES.txt.

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.h [hunk 4]
```diff
@@ -24,6 +24,7 @@
 extern void* MainMenuInit(bool);
 extern int MainMenuRun(void);
 
+
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.h:27]: Unneccesary change.
```suggestion
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 5]
```diff
@@ -29,58 +30,59 @@
...
 
 
 static void
-onPlayerConfigMenuActivate(void * /* dummy */)
+onPlayerConfigMenuActivate(void* /* dummy */)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:37]: Rename function to reflect the skipping of the 'player' config.

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/optionsmenu.cpp [hunk 6]
```diff
@@ -109,18 +77,11 @@ OptionsMenuInit(void *prevMenu)
...
-//#endif
-
-    GfuiMenuCreateButtonControl(MenuHandle, param, "opengl", NULL, onOpenGLMenuActivate);
     GfuiMenuCreateButtonControl(MenuHandle, param, "sound", NULL, onSoundMenuActivate);
-    GfuiMenuCreateButtonControl(MenuHandle, param, "simulation", NULL, onSimuMenuActivate);
-    GfuiMenuCreateButtonControl(MenuHandle, param, "ai", NULL, onAIMenuActivate);
     GfuiMenuCreateButtonControl(MenuHandle, param, "back", prevMenu, GfuiScreenActivate);
 
     GfParmReleaseHandle(param);
+    //SIMULATED DRIVING ASSISTANCE: removed graphics, advanced graphics, opengl, simulation, ai opponents menu
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/optionsmenu.cpp:84]: Move this to the top of the function

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp [hunk 7]
```diff
@@ -1,27 +1,27 @@
...
+  *   the Free Software Foundation; either version 2 of the License, or     *
+  *   (at your option) any later version.                                   *
+  *                                                                         *
+  ***************************************************************************/
+
+  /** @file
+			  Race selection menu
+	  @author	<a href=mailto:eric.espie@torcs.org>Eric Espie</a>
+	  @version	$Id: raceselectmenu.cpp 7011 2020-04-30 15:23:52Z iobyte $
+  */
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp:24]: Don't run the automatic linter on a speed-dreams file, revert this and then add your changes.

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp [hunk 8]
```diff
@@ -400,46 +399,16 @@ RmStopRaceMenu()
...
     screen[i++] = rmQuitHookInit();
 
+    // SIMULATED DRIVING ASSISTANCE: removed controls and forcefeedback menu's
     rmStopScrHandle = rmStopRaceMenu(buttonRole[0], screen[0],
                buttonRole[1], screen[1],
                buttonRole[2], screen[2],
                buttonRole[3], screen[3],
-               buttonRole[4], screen[4],
-               buttonRole[5], screen[5],
+               0, 0,
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp:410]: use `nullptr`, because all of these values are actually pointers.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 9]
```diff
@@ -312,13 +332,41 @@ Changed:
...
-        added controls for intervention sound volume
-    controlconfigmenu.xml
-        added controls for intervention toggle on/off
+  data/
+      config/raceman
+        CMakeLists.txt
+        quickrace.xml
+            changed standard track
+        practice.xml
+            removed
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:341]: Something like this would be more clear and save space, do this for all removed files.
```suggestion
        removed practice.xml
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 10]
```diff
@@ -312,13 +332,41 @@ Changed:
...
+        singleevent-challange.xml
+            removed
+        singleeven-endurance.xml
+            removed
+      data/
+        credits.xml
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:347]: State what you updated.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 11]
```diff
@@ -187,27 +187,47 @@ Changed:
...
                     added    ResearcherMenu.h and .cpp
                     added    DataSelectionMenu.h and .cpp
                     added    backend in both ADD_SDLIB_LIBRARY and ADD_SDLIB_INCLUDEDIR
-                mainscreens/mainmenu.cpp
-                    included "DataSelectionMenu.h"
-                    added    GoBack()
-                    changed  MainMenuInit()
+                    removed player, graph, opengl, advanced, simu, ai, hostsettings configs
+                mainscreens
+                    /mainmenu.cpp
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:198]: Small detail, but this is how we have been formatting the directories, so change that for consistency (also update this for all your other changes)
```suggestion
                mainscreens/
                    mainmenu.cpp
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 12]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                confscreens
+                    /soundconfig.cpp
                         added ChangeInterventionVolume()
                         added logic to create new controls for intervention volume and store their data
                         changed readSoundCfg()
-                    controlconfig.cpp
-                        added cmd info for intervention toggle on/off to the Cmd[] member variable.
+                    /controlconfig 
+                        removed parameters in init
+                    removed player, graph, opengl, advanced, simu, ai, hostsettings config .h/.cpp
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:229]: For clarity
```suggestion
                    removed .h and .cpp files: player, graph, opengl, advanced, simu, ai, hostsettings config
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Main comments:
- Don't run the automatic formatter on speed-dreams files. Please revert this and add your changes to the orginal files instead.
- Crash: Restart race in RaceStopMenu crashes the game.
- Crash: Abandoning the race in RaceStopMenu, and then Starting it again, crashes the game.
- In the 'RaceSelectMenu' there is still a button saying 'Players', however this button brings you to the Controls menu, since the playersconfig menu was removed, so this label should be updated.

File: speed-dreams/source-2.2.3/data/config/raceman/CMakeLists.txt [hunk 13]
```diff
@@ -1,7 +1,7 @@
 INCLUDE(../../../cmake/macros.cmake)
 
-SET(_RACE_FILES practice.xml quickrace.xml
-    singleevent-challenge.xml singleevent-endurance.xml)
+SET(_RACE_FILES quickrace.xml
+    )
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/data/config/raceman/CMakeLists.txt:4]:
```suggestion
SET(_RACE_FILES quickrace.xml)
```

File: speed-dreams/source-2.2.3/data/data/menu/mainmenu.xml [hunk 14]
```diff
@@ -24,11 +24,11 @@
 <params name="mainmenu">
 
 	<section name="dynamic controls">
-
-	<section name="race">
+    <!--// SIMULATED DRIVING ASSISTANCE: FROM RACE TO START-->
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/data/data/menu/mainmenu.xml:27]:
```suggestion
    <!-- SIMULATED DRIVING ASSISTANCE: FROM RACE TO START-->
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 15]
```diff
@@ -20,63 +20,65 @@
...
 #include "mainmenu.h"
 #include "exitmenu.h"
 #include "optionsmenu.h"
 #include "creditsmenu.h"
 #include "DataSelectionMenu.h"
 
-
-static void *MenuHandle = 0;
+// SIMULATED DRIVING ASSISTANCE: RENAMED MenuHandle
+static void *MainMenuHandle = 0;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:33]: Was there a technical reason for this rename?

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp (hunk 15 above)
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:33]: well I first thought I needed it as a global variable, but I fixed it without that, so it not necessary anymore. I should porbably revert that

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/optionsmenu.cpp [hunk 16]
```diff
@@ -21,16 +21,12 @@
...
-#include <openglconfig.h>
 #include <soundconfig.h>
-#include <simuconfig.h>
-#include <aiconfig.h>
 
-#include "optionsmenu.h"
 
+#include "optionsmenu.h"
 
+ // SIMULATED DRIVING ASSISTANCE: REMOVED player, graph, opengl, advanced, simu, ai, hostsettings configs
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/optionsmenu.cpp:29]:
```suggestion
 // SIMULATED DRIVING ASSISTANCE: removed player, graph, opengl, advanced graph, simu, ai, hostsettings configs
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp [hunk 17]
```diff
@@ -38,16 +38,15 @@
...
-#endif
+#include <controlconfig.h>
+
 
 #include "legacymenu.h"
 #include "racescreens.h"
 
+#include "mainmenu.h"
+
+// SIMULATED DRIVING ASSISTANCE: removed networking
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp:49]: no need for a comment for the #includes. Just putting it in the CHANGES.txt is enough

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp [hunk 18]
```diff
@@ -460,8 +365,16 @@ rmResumeRace(void * /* dummy */)
 	LmRaceEngine().resumeRace();
 }
 
-#ifndef CLIENT_SERVER
-// Init. function for the current menu -----------------------------------------------------
+static void BackToMainMenu(void* p_prevMenu)
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp:368]: missing documentation AND change comment

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp [hunk 19]
```diff
@@ -536,11 +413,11 @@ RmRacemanMenu()
 	GfuiMenuCreateButtonControl(ScrHandle, menuXMLDescHdle, "ConfigureRaceButton",
 								NULL, RmConfigureRace);
 	if (SupportsHumanDrivers)
-		GfuiMenuCreateButtonControl(ScrHandle, menuXMLDescHdle, "ConfigurePlayersButton",
-								NULL, rmOnPlayerConfig);
+		GfuiMenuCreateButtonControl(ScrHandle, menuXMLDescHdle, "ControlsButton",
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp:416]: missing change comment

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp [hunk 20]
```diff
@@ -112,6 +91,33 @@ rmOnSelectRaceMan(void *pvRaceManTypeIndex)
...
+	GfuiScreenActivate(prevMenu);
+
+	LmRaceEngine().cleanup();
+
+	LegacyMenu::self().shutdownGraphics(/*bUnloadModule=*/true);
+}
+
+
+
+
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp:120]: - remove excess whitespace
- missing change comments
- functions we added to SD need to follow our guidelines (just SELECT this and run clang-format, do NOT run it on the whole file)

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp [hunk 21]
```diff
@@ -122,7 +128,7 @@ void *
 RmRaceSelectInit(void *prevMenu)
 {
     if (RmRaceSelectMenuHandle) 
-		return RmRaceSelectMenuHandle;
+            return RmRaceSelectMenuHandle;
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp:132]: Unnecessary

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp [hunk 22]
```diff
@@ -124,49 +123,8 @@ rmRestartRaceHookInit()
...
-static void *
-rmControlsHookInit()
-{
-    if (!pvControlsHookHandle)
-        pvControlsHookHandle = GfuiHookCreate(0, rmControlsHookActivate);
-
-    return pvControlsHookHandle;
-}
 
+//SIMULATED DRIVING ASSISTANCE: remove controls setting
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp:127]:
```suggestion
//SIMULATED DRIVING ASSISTANCE: removed controls settings
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp [hunk 23]
```diff
@@ -409,7 +362,7 @@ RmStopRaceMenu()
 
     // Attempt to find a human driver
     for (j=0; ; j++) {
-        snprintf(buf, sizeof(buf), "%s/%s/%d", ROB_SECT_ROBOTS, ROB_LIST_INDEX, j+1);
+        snprintf(buf, sizeof(buf), "%s/%s/%d", ROB_SECT_ROBOTS, ROB_LIST_INDEX, j + 1);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racestopmenu.cpp:365]: Unnecessary change
```suggestion
        snprintf(buf, sizeof(buf), "%s/%s/%d", ROB_SECT_ROBOTS, ROB_LIST_INDEX, j+1);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 24]
```diff
@@ -187,27 +187,47 @@ Changed:
...
             legacymenu/
                 CMakeList.txt
                     added    ResearcherMenu.h and .cpp
                     added    DataSelectionMenu.h and .cpp
                     added    backend in both ADD_SDLIB_LIBRARY and ADD_SDLIB_INCLUDEDIR
-                mainscreens/mainmenu.cpp
-                    included "DataSelectionMenu.h"
-                    added    GoBack()
-                    changed  MainMenuInit()
+                    removed player, graph, opengl, advanced, simu, ai, hostsettings configs
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:196]:
```suggestion
                    removed player, graph, opengl, advanced graph, simu, ai, hostsettings configs
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 25]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                mainscreens/
+                    mainmenu.cpp
+                        included "DataSelectionMenu.h"
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:199]:
```suggestion
                        included "DataSelectionMenu.h"
                        removed  <playerconfig.h>
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 26]
```diff
@@ -20,26 +20,27 @@
...
 #include "DataSelectionMenu.h"
 
-
 static void *MenuHandle = 0;
 
 
 static void
-onPlayerConfigMenuActivate(void * /* dummy */)
+onControlMenuActivate(void * /* dummy */)
 {
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:37]: add change comment

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 27]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                        added    GoBack()
+                        changed  MainMenuInit()
+                        changed  onplayerconfigMenuActivate()
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:202]:
```suggestion
                        changed  onplayerconfigMenuActivate(): renamed it to onControlMenuActivate()
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 28]
```diff
@@ -118,10 +119,11 @@ MainMenuInit(bool SupportsHumanDrivers)
     GfuiMenuCreateStaticControls(MenuHandle, menuDescHdle);
 
     //Add buttons and create based on xml
-    GfuiMenuCreateButtonControl(MenuHandle, menuDescHdle, "race", NULL, onRaceSelectMenuActivate);
+    // SIMULATED DRIVING ASSISTANCE: FROM RACE TO START
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:122]:
```suggestion
    // SIMULATED DRIVING ASSISTANCE: changed button name from race to start
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 29]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                        changed  onRaceSelectActivate()
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:203]: onRaceSelectActivate() doesn't even exist and onRaceSelectMenuActivate() has not been changed.
MainMenuInit(), however, has been changed
```suggestion
                        changed  MainMenuInit()
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 30]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                    optionsmenu.cpp
+                        removed graphics, advanced graphics, opengl, simulation, ai opponents menu
                 racescreens/
                     racerunningmenus.cpp
                         removed RmProgressiveTimeModifier implementation and calls
                     racestopmenu.cpp
                         removed rmProgressiveTimeModifier.start() call
+                        removed controlsmenu 
+                    racemanmenu.cpp
+                        removed networking and go back to main menu
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:213]:
```suggestion
                    racemanmenu.cpp
                        included <controlconfig.h> and "mainmenu.h"
                        removed <playerconfig.h>, <csnetwork.h> and <network.h>
                        removed networking and go back to main menu
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 31]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                        changed rmOnRaceDataChanged
+                        added BackToMainMenu()
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:215]:
```suggestion
                        changed rmOnRaceDataChanged()
                        changed rmOnPlayerConfig(): renamed it to rmOnControlConfig()
                        added   BackToMainMenu()
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 32]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                        changed rmRaceManMenu()
+                    racescreens.h
+                        removed networking
+                    raceselectmenu.cpp
+                        skip through this in rmOnActivate()
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:220]:
```suggestion
                        changed rmOnActivate()
                        changed rmOnRaceSelectShutdown()
                        changed rmOnSelectRaceMan()
                        changed RmRaceSelectInit()
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 33]
```diff
@@ -187,27 +187,47 @@ Changed:
...
+                    removed csnetclientsettings.cpp, csnetserversettings.cpp ,csnetworkingmenu.cpp, networking.cpp, csnetwork.h
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:221]: Consider adding a removed section between the added and changed sections
```suggestion
                    removed csnetclientsettings.cpp, csnetserversettings.cpp, csnetworkingmenu.cpp, networking.cpp, csnetwork.h
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt [hunk 34]
```diff
@@ -187,27 +187,47 @@ Changed:
...
                 confscreens/
                     soundconfig.cpp
                         added ChangeInterventionVolume()
                         added logic to create new controls for intervention volume and store their data
                         changed readSoundCfg()
-                    controlconfig.cpp
-                        added cmd info for intervention toggle on/off to the Cmd[] member variable.
+                    controlconfig
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/CHANGES.txt:227]: this is only for the .h, missing all changes int the .cpp

**Review by shoekzema — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/.gitignore [hunk 35]
```diff
@@ -36,4 +36,6 @@ Folder.DotSettings.user
 
 temp/
 
-RppUtilsCoverage.bat
\ No newline at end of file
+RppUtilsCoverage.bat
+
+enc_temp_folder
```
[the remark this answers is not in the document]
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/.gitignore:41]: i did see it on gitkraken once, but maybe i was still building

**Review by suppergerrie2 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp (hunk 20 above)
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/raceselectmenu.cpp:121]: in only added line 104, dont know why it says i changed more

File: speed-dreams/source-2.2.3/data/data/menu/stopracemenu.xml [hunk 36]
```diff
@@ -30,17 +30,14 @@
...
 		<!-- button texts and tips -->
 		<attstr name="restart.text" val="Restart Race"/>
 		<attstr name="restart.tip" val="Restart the current race"/>
 		<attstr name="abort.text" val="Abandon Race"/>
 		<attstr name="abort.tip" val="Abort the current race"/>
 		<attstr name="resume.text" val="Resume Race"/>
 		<attstr name="resume.tip" val="Return to the race"/>
-		<attstr name="controls.text" val="Adjust Controls"/>
-		<attstr name="controls.tip" val="Adjust the Player's Controls"/>
-		<attstr name="forcefeedback.text" val="Adjust Force Feedback"/>
```
[the remark this answers is not in the document]
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/data/data/menu/stopracemenu.xml:42]: Its back

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp [hunk 37]
```diff
@@ -20,26 +20,26 @@
...
 #include "DataSelectionMenu.h"
 
-
 static void *MenuHandle = 0;
 
-
+// SIMULATED DRIVING ASSISTANCE: renamed from onPlayerConfigMenuActivate
 static void
-onPlayerConfigMenuActivate(void * /* dummy */)
+onControlMenuActivate(void * /* dummy */)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:35]: Currently the controls menu is accesible via the main menu and the racestart menu. It would probably be best to simplify this even more and to just remove this entirely from the main menu. Less path to take through the menu's is always better :)

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp [hunk 38]
```diff
@@ -460,8 +364,20 @@ rmResumeRace(void * /* dummy */)
 	LmRaceEngine().resumeRace();
 }
 
-#ifndef CLIENT_SERVER
-// Init. function for the current menu -----------------------------------------------------
+// SIMULATED DRIVING ASSISTANCE: added function to go back to main menu
+/// @brief				goes back to the main menu, cleans up the race engine and shutsdown the graphics,
+///						according to the raceselectmenu that we skip
```
[the remark this answers is not in the document]
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/racescreens/racemanmenu.cpp:369]: why are tabs back :,<

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp (hunk 37 above)
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/mainmenu.cpp:35]: the racestart menu will probably be removed or drastically changed later on, when we have for example track selection in the research menu. Therefore it does not really matter at this point.

**Review by larsdekwant — APPROVED** [review]

**Review by shoekzema — CHANGES_REQUESTED** [review]

**Review by shoekzema — APPROVED** [review]
